doc/manual/package.nix

```diff
@@ -0,0 +1,61 @@
+{ lib
+, mkMesonDerivation
+
+, meson
+, ninja
+, lowdown
+, mdbook
+, mdbook-linkcheck
+, jq
+, nix-ng
+
+# Configuration Options
+
+, version
+}:
+
+let
+  inherit (lib) fileset;
+in
+
+mkMesonDerivation (finalAttrs: {
+  pname = "nix-internal-api-docs";
+  inherit version;
+
+  workDir = ./.;
+  fileset = fileset.unions [
+    ./.version
+    ./meson.build
+    ../../.version
+    ../../doc/manual
+  ];
+
+  # Hack for sake of the dev shell
+  passthru.baseNativeBuildInputs = [
+    meson
+    ninja
+    (lib.getBin lowdown)
+    mdbook
+    mdbook-linkcheck
+    jq
+  ];
+
+  nativeBuildInputs = finalAttrs.passthru.baseNativeBuildInputs ++ [
```

Comment:
thought: This could be factored out into `mkMesonDerivation`, or one of its layers, or maybe it should be called `mkNixMesonDerivation` or something because it's getting a bit specific.